### ISSUE TITLE
1.3b0, isobuild conflict with registerBuildPlugin (having npmDependencies?)

### ISSUE THREAD

**Issue by jiku** [meteor/meteor#5799]
After updating to 1.3b0 from 1.2.1 I'm having issues with isobuild compiling 'unusual' files in `client/` (.md and .cjsx in my case). Seems related to `Package.registerBuildPlugin` (with `npmDependencies`) in [`ccorcos:cjsx`](https://github.com/ccorcos/meteor-cjsx/blob/master/package.js) and [`lalomartins:markd`](https://github.com/lalomartins/meteor-markd/blob/master/package.js)? (links to package.js). Simple repro [here](https://github.com/jiku/meteor-modules).

Example output
```
=> Errors prevented startup:

   While building the application:
   /Users/jiku/app/.meteor/packages/meteor-tool/.1.1.11-modules.0.1uwqwk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/isobuild/compiler-deprecated-compile-step.js:355:25: self is not defined (compiling
   client/sections/about/about.md)
   at Object.compileStep.addJavaScript (/Users/jiku/app/.meteor/packages/meteor-tool/.1.1.11-modules.0.1uwqwk++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/isobuild/compiler-deprecated-compile-step.js:355:25)
   at Package (<runJavaScript-103>:57:15)
```

**Comment by laurentpayot** [comment]
Same here.

**Comment by benjamn** [comment]
Should be fixed in 1.3b1.

**Comment by jiku** [comment]
With the [repro](https://github.com/jiku/meteor-modules), doing `meteor update --release=1.3-modules-beta.1`

outputs

```
=> Errors while initializing project:

While building package lalomartins:markd:
/tools/isobuild/compiler-deprecated-compile-step.js:364:45: Cannot read property 'fileOptions' of undefined (compiling README.md)
at Object.compileStep.addJavaScript (/tools/isobuild/compiler-deprecated-compile-step.js:364:45)
at Package (packages/compileMarkdownFiles/plugin/compile-md.js:41:1)
```

Removing the README.md from that package, meteor update outputs

```
=> Errors prevented startup:

   While building the application:
   client/test.md: Markdown compiler error: Unknown htmljs type: Tag
```

Not sure about the last one but with the first, I'm hoping that won't happen in every case where there's a README.md at the package root. :)

**Comment by mquandalle** [comment]
I’m experiencing the same bug as @jiku.

**Comment by avital** [comment]
This should be fixed in `1.3-modules-beta.2`, can you help verify?

**Comment by jiku** [comment]
Yeah b2 seems to fix these (now having #5871 / #5879).
